tf-psa-crypto/drivers/builtin/include/mbedtls/ecdh.h

```diff
@@ -141,6 +152,47 @@ typedef struct mbedtls_ecdh_context {
 }
 mbedtls_ecdh_context;
 
+#if defined(MBEDTLS_ECDH_LEGACY_CONTEXT)
+#if defined(MBEDTLS_ECP_RESTARTABLE)
+#define MBEDTLS_ECDH_CONTEXT_INIT { MBEDTLS_ECP_GROUP_INIT, MBEDTLS_MPI_INIT, \
+                                    MBEDTLS_ECP_POINT_INIT, \
+                                    MBEDTLS_ECP_POINT_INIT, MBEDTLS_MPI_INIT, \
+                                    MBEDTLS_ECP_PF_UNCOMPRESSED, \
+                                    MBEDTLS_ECP_POINT_INIT, MBEDTLS_ECP_POINT_INIT, \
+                                    MBEDTLS_MPI_INIT, 0, \
+                                    MBEDTLS_ECP_RESTART_INIT }
+#else
+#define MBEDTLS_ECDH_CONTEXT_INIT { MBEDTLS_ECP_GROUP_INIT, MBEDTLS_MPI_INIT, \
+                                    MBEDTLS_ECP_POINT_INIT, \
+                                    MBEDTLS_ECP_POINT_INIT, MBEDTLS_MPI_INIT, \
+                                    MBEDTLS_ECP_PF_UNCOMPRESSED, \
+                                    MBEDTLS_ECP_POINT_INIT, MBEDTLS_ECP_POINT_INIT, \
+                                    MBEDTLS_MPI_INIT }
+#endif /* MBEDTLS_ECP_RESTARTABLE */
+#else
+#if !defined(MBEDTLS_ECDH_VARIANT_EVEREST_ENABLED)
+#if defined(MBEDTLS_ECP_RESTARTABLE)
+#define MBEDTLS_ECDH_CONTEXT_INIT { MBEDTLS_ECP_PF_UNCOMPRESSED, MBEDTLS_ECP_DP_NONE, \
+                                    MBEDTLS_ECDH_VARIANT_NONE, \
+                                    { .mbed_ecdh = MBEDTLS_ECDH_CONTEXT_MBED_INIT }, 0 }
```

Comment:
We are currently using [designated initializers](https://en.cppreference.com/w/c/language/initialization) (a C99 construct) in library code, but not in public headers. We want public headers to be compatible with C++ as well as C. In C++, [designated initializers](https://en.cppreference.com/w/cpp/language/aggregate_initialization#Designated_initializers) were only made available in C++20 (though many compilers already supported them in non-strict mode).

I'm not sure if it's ok to start requiring C++20.

Anyway using a designated initializer here is not really useful. We don't know which member of the union will eventually be used, since it depends on the curve. So we might as well initialize the first member. Especially when Everest is disabled and the union has a single member. But we can do that even when Everest is enabled! The member `mbed_ecdh` is always present, to handle curves that Everest doesn't support.